packages/store/ts/config/typeExtensions.ts

```diff
@@ -0,0 +1,19 @@
+import { z } from "zod";
+import { EnumsConfig, StoreSimpleOptions, TablesConfig, zStoreConfig } from "./parseStoreConfig";
+
+// Inject non-generic options into the core config.
+// Re-exporting an interface of an existing module merges them, adding new options to the interface.
+// (typescript has no way to override types)
+declare module "@latticexyz/config" {
+  // Extend the user config type, which represents the config as written by the users.
+  // Most things are optional here.
+  export interface MUDCoreUserConfig extends StoreSimpleOptions, EnumsConfig<string> {
+    tables: TablesConfig<string, string>;
+  }
+
+  // Also extend the config type, which represents the configuration after it has been resolved.
+  // It should not have any optional properties, with the default values applied instead.
+  // Other plugins may receive this resolved config as their input.
```

Comment:
Do they always?
```suggestion
  // Other plugins receive this resolved config as their input.
```

Reply:
I think I meant that the plugins that come before don't receive that particular config as input, but otherwise they always do yeah